### PR TITLE
feat(sbb-datepicker): parse ISO dates

### PR DESCRIPTION
**Pull request by MarioCastigliano**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/master/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes [#1753](https://github.com/lyne-design-system/lyne-components/issues/1753)

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

-

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [x] Edge Desktop
- [ ] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: src/components/sbb-datepicker/sbb-datepicker.tsx [hunk 1]
```diff
@@ -253,6 +263,28 @@ export class SbbDatepicker implements ComponentInterface {
...
+      return this._formatValue(value);
+    }
+
+    if (!Number.isNaN(+value) && Number.isInteger(+value)) {
+      const d = new Date(+value);
+      return this._composeValueString(`${d.getDate()}`, `${d.getMonth() + 1}`, d.getFullYear());
+    } else if (this._dateAdapter.isValid(new Date(value))) {
+      const d = new Date(value);
+      return this._composeValueString(`${d.getDate()}`, `${d.getMonth() + 1}`, d.getFullYear());
+    }
```
Comment (DavideMininni-Fincons) [inline src/components/sbb-datepicker/sbb-datepicker.tsx:283]: The difference between the two is just the _+_ in the `new Date()`, right? Maybe a method can be extracted?

File: src/components/sbb-datepicker/sbb-datepicker.tsx [hunk 2]
```diff
@@ -253,6 +268,26 @@ export class SbbDatepicker implements ComponentInterface {
...
+      return '';
+    }
+
+    const match: RegExpMatchArray = value.match(FORMAT_DATE);
+
+    if (match && match?.index === 0) {
+      return this._formatValue(value);
+    }
+
+    if (!Number.isNaN(+value) && Number.isInteger(+value)) {
```
Comment (jeripeierSBB) [inline src/components/sbb-datepicker/sbb-datepicker.tsx:282]: I think this check is enough as NaN is not an Integer?
```suggestion
    if (Number.isInteger(+value)) {
```

File: src/components/sbb-datepicker/sbb-datepicker.tsx [hunk 3]
```diff
@@ -253,6 +268,26 @@ export class SbbDatepicker implements ComponentInterface {
...
   }
 
+  private _getValidValue(value: string): string {
+    if (!value) {
+      return '';
+    }
+
+    const match: RegExpMatchArray = value.match(FORMAT_DATE);
+
+    if (match && match?.index === 0) {
```
Comment (jeripeierSBB) [inline src/components/sbb-datepicker/sbb-datepicker.tsx:278]:
```suggestion
    if (match?.index === 0) {
```

**Review by jeripeierSBB — CHANGES_REQUESTED** [review]

File: src/components/sbb-datepicker/sbb-datepicker.tsx [hunk 4]
```diff
@@ -189,6 +189,9 @@ export class SbbDatepicker implements ComponentInterface {
   public connectedCallback(): void {
     this._handlerRepository.connect();
     this._inputElement = getInput(this._element, this.input);
+    if (this._inputElement) {
+      this._inputElement.value = this._getValidValue(this._inputElement?.value);
```
Comment (kyubisation) [inline src/components/sbb-datepicker/sbb-datepicker.tsx:193]:
```suggestion
      this._inputElement.value = this._getValidValue(this._inputElement.value);
```

File: src/components/sbb-datepicker/sbb-datepicker.tsx [hunk 5]
```diff
@@ -253,6 +268,26 @@ export class SbbDatepicker implements ComponentInterface {
...
+      return '';
+    }
+
+    const match: RegExpMatchArray = value.match(FORMAT_DATE);
+
+    if (match?.index === 0) {
+      return this._formatValue(value);
+    }
+
+    if (Number.isInteger(+value)) {
```
Comment (kyubisation) [inline src/components/sbb-datepicker/sbb-datepicker.tsx:282]: Nitpick
```suggestion
    } else if (Number.isInteger(+value)) {
```

**Review by kyubisation — APPROVED** [review]
Generally looks good to me, thank you for your work 👍 
One thing I am not sure about is server side rendering; If the date is passed as an ISO string, is it reformated quickly enough when transitioning to the client or would the client see the ISO string, before it is reformated.

**Review by kyubisation — APPROVED** [review]
LGTM 👍 
Thank you very much!

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**